opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/WebOpacAt.java

```diff
@@ -0,0 +1,294 @@
+        return first == null ? null : first.nextElementSibling().text();
+    }
+
+    private static String getCover(Element doc) {
+        return doc.select(".coverimage img").first().attr("src").replaceFirst("&width=\\d+", "");
+    }
+
+    @Override
+    public SearchRequestResult filterResults(Filter filter, Filter.Option option)
+            throws IOException, OpacErrorException {
+        return null;
+    }
+
+    @Override
+    public DetailledItem getResult(int position) throws IOException, OpacErrorException {
+        // Not necessary since getResultById returns an ID with every result
+        return null;
+    }
+
+    @Override
+    public List<SearchField> getSearchFields()
+            throws IOException, OpacErrorException, JSONException {
+        start();
+        final List<SearchField> fields = new ArrayList<>();
+        addSimpleSearchField(fields);
+        addAdvancedSearchFields(fields);
+        return fields;
+    }
+
+    protected void addSimpleSearchField(List<SearchField> fields)
+            throws IOException, JSONException {
+        final String html = httpGet(opac_url + "/search?mode=s", getDefaultEncoding());
+        final Document doc = Jsoup.parse(html);
+        final Element simple = doc.select(".simple_search").first();
+        final TextSearchField field = new TextSearchField();
+        field.setFreeSearch(true);
+        field.setDisplayName(simple.select("h4").first().text());
+        field.setId(simple.select("#keyboard").first().attr("name"));
+        field.setHint("");
+        field.setData(new JSONObject());
+        field.getData().put("meaning", field.getId());
+        fields.add(field);
+    }
+
+    protected void addAdvancedSearchFields(List<SearchField> fields)
+            throws IOException, JSONException {
+        final String html = httpGet(opac_url + "/search?mode=a", getDefaultEncoding());
+        final Document doc = Jsoup.parse(html);
+        final Elements options = doc.select("select#adv_search_crit_0").first().select("option");
+        for (final Element option : options) {
+            final TextSearchField field = new TextSearchField();
+            field.setDisplayName(option.text());
+            field.setId(option.val());
+            field.setHint("");
+            field.setData(new JSONObject());
+            field.getData().put("meaning", field.getId());
+            fields.add(field);
+        }
+    }
+
+    @Override
+    protected String getDefaultEncoding() {
+        return "utf-8";
+    }
+
+    @Override
+    public String getShareUrl(String id, String title) {
+        return opac_url + "/search?view=detail&id=" + id;
+    }
+
+    @Override
+    public int getSupportFlags() {
+        return SUPPORT_FLAG_ENDLESS_SCROLLING;
+    }
+
+    @Override
+    public Set<String> getSupportedLanguages() throws IOException {
+        return null;
+    }
+
+    @Override
+    public void setLanguage(String language) {
+    }
```

Comment:
You should implement `getSupportedLanguages` and `setLanguage` so that the app can set the language depending on the device language.
`getSupportedLanguages` could do a network operation to find out the available languages, but if it is German, English and Turkish in all the libraries, you could return those codes directly as well.
The current language of the device is passed to `setLanguage`, which you should save and then use for the search and detail requests. If the language passed to `setLanguage` is not one of the supported ones, you should fall back to English (and, if that should also not be supported, German).
See also the [Javadoc](https://github.com/raphaelm/opacclient/blob/master/opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/OpacApi.java#L587) and the implementation in [Pica](https://github.com/raphaelm/opacclient/blob/master/opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/Pica.java#L865)